polaris-react/src/components/TextField/TextField.tsx

```diff
@@ -164,7 +164,7 @@ interface NonMutuallyExclusiveProps {
   /** Visual styling options for the TextField
    * @default 'inherit'
    */
-  variant?: 'inherit' | 'borderless';
+  variant?: 'inherit' | 'borderless' | 'slim' | 'borderlessSlim';
```

Comment:
Lmk if this api choice is ok. We could also create a new boolean prop called `inTable` or something to make the use case more specific and use it with `variant='borderless'`.

|Option 1 (this PR)|Option 2|Option 3|
|-|-|-|
|`<TextField variant='borderlessSlim' />`|`<TextField variant='borderless' inTable />`|Keep custom text field implementation in `Filters`|

Reply:
that makes sense to me! I'll align it to what we have in `Button`, both `variant` and `size`